disnake/mentions.py

```diff
@@ -69,13 +69,26 @@ def __init__(
         self,
         *,
         everyone: bool = default,
-        users: Union[bool, List[Snowflake]] = default,
-        roles: Union[bool, List[Snowflake]] = default,
+        users: Union[bool, Sequence[Snowflake]] = default,
+        roles: Union[bool, Sequence[Snowflake]] = default,
         replied_user: bool = default,
     ) -> None:
         self.everyone = everyone
-        self.users = users
-        self.roles = roles
+        # TODO(3.0): annotate attributes as `Sequence` instead of copying to list
+        self.users: Union[bool, List[Snowflake]]
+        self.roles: Union[bool, List[Snowflake]]
+        if users is default:
+            self.users = users  # type: ignore
+        elif isinstance(users, bool):
+            self.users = users
+        else:
+            self.users = list(users)
+        if roles is default:
+            self.roles = roles  # type: ignore
+        elif isinstance(roles, bool):
+            self.roles = roles
+        else:
+            self.roles = list(roles)
```

Comment:
could simplify it a little?
```suggestion
        if users is default or isinstance(users, bool):
            self.users = cast("bool", users)
        else:
            self.users = list(users)
        if roles is default or isinstance(roles, bool):
            self.roles = cast("bool", roles)
        else:
            self.roles = list(roles)
```